app/src/main/kotlin/com/hippo/ehviewer/coil/CropBorderInterceptor.kt

```diff
@@ -20,6 +35,25 @@ object CropBorderInterceptor : Interceptor {
                     val (x, y) = image.rect.topLeft
                     val (w, h) = image.rect.size
                     val src = image.image.bitmap
+
+                    // Large hardware bitmaps have rendering issues (e.g. crash, empty) on some devices.
+                    // This is not ideal but I haven't figured out how to probe the threshold.
+                    // All we know is that it's less than the maximum texture size.
+                    if (isAtLeastQ && maxOf(w, h) <= chain.request.hardwareThreshold) {
+                        runSuspendCatching {
+                            resourceScope {
+                                val buffer = autoCloseable { HardwareBuffer.create(w, h, FORMAT, 1, USAGE) }
+                                copyBitmapToAHB(src, buffer, x, y)
+                                Bitmap.wrapHardwareBuffer(buffer, chain.request.colorSpace)
+                            }
+                        }.onSuccess { bitmap ->
+                            bitmap?.let {
+                                src.recycle()
+                                return result.copy(image = image.copy(image = bitmap.asImage(), rect = image.rect.size.toIntRect()))
```

Comment:
Maybe use `var bitmap = null` to avoid repeating `src.recycle()` & `result.copy()`

Reply:
Also when will it fails? Maybe we should log the error.

Reply:
Done